app/Http/Controllers/Twill/EventController.php

```diff
@@ -2,49 +2,56 @@
+use A17\Twill\Services\Listings\TableColumns;
+use App\Models\Event;
 use App\Repositories\EventProgramRepository;
 
-class EventController extends \App\Http\Controllers\Twill\ModuleController
+class EventController extends BaseController
 {
-    protected $moduleName = 'events';
-    protected $previewView = 'site.events.detail';
-
-    protected $indexOptions = [
-        'publish' => true,
-        'bulkPublish' => true,
-        'duplicate' => true,
-    ];
-
-    protected $indexColumns = [
-        'image' => [
-            'thumb' => true,
-            'optional' => false,
-            'variant' => [
-                'role' => 'hero',
-                'crop' => 'default',
-            ],
-        ],
-        'title' => [
-            'title' => 'Title',
-            'field' => 'title',
-        ],
-        'formattedNextOccurrence' => [
-            'title' => 'Event Date',
-            'field' => 'formattedNextOccurrence',
-            'present' => true,
-            'sort' => true,
-        ]
-    ];
-
-    protected $featureField = 'landing';
-
-    protected $indexWith = ['medias'];
-
-    protected $formWith = ['revisions', 'dateRules'];
-
-    protected $filters = [];
-
-    protected $defaultOrders;
+    protected function setUpController(): void
+    {
+        $this->eagerLoadFormRelations(['revisions', 'dateRules']);
+        $this->eagerLoadListingRelations(['medias']); // I don't believe this is necessary anymore
+        $this->enableBulkFeature();
+        $this->enableDuplicate();
+        $this->enableFeature();
+        $this->enableShowImage();
+        $this->setFeatureField('landing');
+        $this->setModuleName('events');
+        $this->setPreviewView('site.events.detail');
+        parent::setUpController();
+    }
+
+    protected function additionalIndexTableColumns(): TableColumns
+    {
+        $columns = new TableColumns();
+        $columns->add(
+            Text::make()
+                ->field('formattedNextOccurrence')
+                ->title('Event Date')
+                ->sortable()
+                ->customRender(self::renderDate(...))
+        );
+
+        return $columns;
+    }
+
+    private function renderDate(Event $event)
+    {
+        if (!empty($event->forced_date)) {
+            return $event->forced_date;
+        }
+
+        if ($next = $event->nextOccurrenceExclusive) {
+            return '<time datetime="' . $next->date->format('c') . '" itemprop="startDate">' . $next->date->format('F j, Y | g:i') . '</time>&ndash;<time datetime="' . $next->date_end->format('c') . '" itemprop="endDate">' . $next->date_end->format('g:i') . '</time>';
+        }
+
+        if ($last = $event->lastOccurrence) {
+            return '<time datetime="' . $last->date->format('c') . '" itemprop="startDate">' . $last->date->format('F j, Y | g:i') . '</time>&ndash;<time datetime="' . $last->date_end->format('c') . '" itemprop="endDate">' . $last->date_end->format('g:i') . '</time>';
+        }
```

Comment:
You might be able to use `EventPresenter->formattedNextOccurrence()` in place of this. Check to make sure the logic is same between this code and that method.

Reply:
I've updated the controller to use the existing presenter method.